### PR TITLE
Remove Coupon Code gem and set invite code = id

### PR DESCRIPTION
**Pull request by ankonym**
This does two things:

a) it removes the Coupon Code gem and replaces it as suggested by Elijah

b) it sets the id = the invite code, which will help with invite code deletion after the leap platform tests run.

### CODE REVIEW

File: lib/tasks/invite_code.rake [hunk 1]
```diff
@@ -1,16 +1,22 @@
-
+require 'base64'
+require 'securerandom'
```
Comment (azul) [inline lib/tasks/invite_code.rake]: is this needed?
I think if the InviteCode.new needs it it should go into the invite code model. And it is there already.

File: lib/tasks/invite_code.rake [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 
 desc "Generate a batch of invite codes"
-task :generate_invites, [:n] => :environment do |task, args|
+task :generate_invites, [:n, :u] => :environment do |task, args|
```
Comment (azul) [inline lib/tasks/invite_code.rake]: This does not seem related to the pull request. It probably makes sense anyway - just making sure it's what you want.

File: lib/tasks/invite_code.rake (hunk 1 above)
Comment (ankonym) [inline lib/tasks/invite_code.rake]: True, not anymore - that method got moved around and this was forgotten there. I'll remove it.

File: lib/tasks/invite_code.rake (hunk 2 above)
Comment (ankonym) [inline lib/tasks/invite_code.rake]: This is probably something left over from the git merge/rebase mess we cleaned up since yesterday... so the version with both :n and :u is correct and it should stay, but it's actually part of the multi-use invite codes, not this pull request. Not sure how exactly it ended up here... :neutral_face: but yes, it's what should stay.

Comment (ankonym) [inline lib/tasks/invite_code.rake]: Actually, it shouldn't - I have now cleaned up everything related to multi-use invite codes from this pull request because the rake task to generate invite codes didn't work in this branch. Now both PRs should be correct. Sorry about the confusion!

Comment (azul) [inline lib/tasks/invite_code.rake]: did you push? looks like this is still here...

**Comment by varac** [comment]
@azul, @ankonym i lost track what needs to be done here - "This branch has conflicts that must be resolved"

**Comment by ankonym** [comment]
I'm still stuck with git issues there. Now that I'm back from Sweden, would you have time to help me fix that, @azul ? I rebased as you suggested but am still confused what comes next.

**Comment by azul** [comment]
ankonym, please force push it to your github repo
(git push -f).
That should update this pull request.

**Comment by varac** [comment]
@ankonym if you need support, you can also always drop by the office if you like.
